mmseg/datasets/custom.py

```diff
@@ -85,6 +86,32 @@ def __init__(self,
         self.test_mode = test_mode
         self.ignore_index = ignore_index
         self.reduce_zero_label = reduce_zero_label
+        self.custom_classes = False
+
+        # overwrite classes with custom classes
+        if classes:
+            self.custom_classes = True
+            if not isinstance(classes, (tuple, list)):
+                raise ValueError(
+                    f'Unsupported type {type(classes)} of classes.')
+
+            if self.CLASSES:
+                self.old_id_to_new_id = {}
+                matched_classes = 0
+                for i, c in enumerate(self.CLASSES):
+                    if c not in classes:
+                        self.old_id_to_new_id[
+                            i + int(self.reduce_zero_label)] = -1 + int(
+                                self.reduce_zero_label)
```

Comment:
`reduce_zero_label=True` for a dataset, the CLASSES is already reduced, like ADE20k.

Reply:
If I understood correctly, you suggest using `CLASSES` for the labels, and then transforming the background to -1 in LoadAnnotation using `reduce_zero_label` flag. Is this correct?

Reply:
Yes. We only need to handle `reduce_zero_label` in `LoadAnnotation` and `CustomDataset.get_gt_seg_maps()`.